templates/samples.html

```diff
@@ -0,0 +1,229 @@
+                  <a href$="[[link]]" target="_blank" class="demo-link">
+                    <paper-button>View Demo</paper-button>
+                  </a>
+                </template>
+              </div>
+              <iron-icon icon="{{_computeIconId(feature.category)}}"
+                         title="{{feature.category}}"></iron-icon>
+            </div>
+          </paper-material>
+        </li>
+      </template>
+      </ul>
+  {% endverbatim %}
+
+</div>
+</section>
+
+</template>
+
+{% endblock %}
+
+{% block js %}
+<script>
+(function() {
+
+var search = $('.search paper-input');
+
+var t = $('#t');
+t.categories = {{categories|safe}}; // server included.
+t.prevSelectedCategory = null;
+
+t._computeIcon = function(el) {
+  return 'chromestatus:' + (el ? el.dataset.category : '');
+};
+
+t._computeIconId = function(categoryStr) {
+  return 'chromestatus:' + this.categories[categoryStr];
+};
+
+t._computeFeatureLink = function(id) {
+  return '/features/' + id;
+};
+
+t._onSelectCategory = function(e, detail) {
+  // Enable toggling selected item until
+  // https://github.com/PolymerElements/paper-menu/issues/50 is fixed.
+  if (this.selectedCategory === this.prevSelectedCategory) {
+    this.$.papermenu.selected = null;
+    this.prevSelectedCategory = null;
+  } else {
+    this.prevSelectedCategory = this.selectedCategory;
+  }
+
+  this.filter(search.value);
+};
+
+t._onRippleDone = function(e, detail) {
+  // Needed until https://github.com/PolymerElements/paper-ripple/pull/41 is fixed.
+  this.debounce('ripple', function() {
+    t._onSelectCategory();
+  }, 10);
+};
+
+t.filter = function(val) {
+  var features = this.features;
+  if (val) {
+    var regex = new RegExp(val, 'i');
+    features = features.filter(function(feature, idx, array) {
+      return regex.test(feature.name) || regex.test(feature.summary);
+    });
+  }
+
+  // Further refine list based on selected category in menu.
+  if (this.selectedCategory) {
+    var category = this.selectedCategory.textContent.trim();
+    var regex = new RegExp(category, 'i');
+    features = features.filter(function(feature, idx, array) {
+      return regex.test(feature.category);
+    });
+  }
+
+  this.filtered = features;
+};
+
+t.removeFilters = function() {
+  t.filter();
+};
+
+// Fire of samples.json XHR right away so data can populate faster.
+var samplesPromise = new Promise(function(resolve, reject) {
+  var url = location.hostname == 'localhost' ? 'https://polymer10-samples-dot-cr-status.appspot.com/samples.json' :
+                                               '/samples.json';
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url);
+  xhr.responseType = 'json';
+  xhr.onloadend = function(e) {
+    if (e.target.status < 400) {
+      resolve(e.target.response);
+    } else {
+      reject(new Error('Samples XHR failed with status ' + e.target.statusText));
+    }
+  };
+  xhr.onerror = reject;
+  xhr.send();
+});
+
+samplesPromise.then(function(features) {
```

Comment:
The tl;dr is that it's fine to do this rewriting if you'd like, but there's a historical reason why some samples link to the repo.

Some background: early on, I tended to link to the GitHub repo itself rather than the deployed version on GitHub Pages, since I wanted to direct users to the source code, and there's a link to the GitHub Pages URL in each directory's `README.md`.

Later on, we added "View Source" and "View on GitHub" links to the samples template, so it should be obvious even for inexperienced developers how to get to the underlying source code. At that point, I started linking to the GitHub Pages URL.

If the goal is to always point to the GitHub Pages URL, I should address that by cleaning up the older links in the App Engine datastore. I can do that one at a time as the older samples get migrated to the new templates.